packages/shared/lib/services/proxy.service.ts

```diff
@@ -478,149 +408,27 @@ See https://docs.nango.dev/guides/proxy#proxy-requests for more information.`
         return safeHeaders;
     }
 
-    /**
-     * Get
-     * @param {Response} res Express response object
-     * @param {NextFuncion} next callback function to pass control to the next middleware function in the pipeline.
-     * @param {string} url
-     * @param {ApplicationConstructedProxyConfiguration} config
-     */
-    private async get(
-        url: string,
-        config: ApplicationConstructedProxyConfiguration,
-        activityLogId: number,
-        environment_id: number,
-        options: AxiosRequestConfig
-    ) {
-        try {
-            const headers = this.constructHeaders(config);
-
-            const response: AxiosResponse = await backOff(
-                () => {
-                    return axios.get(url, { ...options, headers });
-                },
-                { numOfAttempts: Number(config.retries), retry: this.retry.bind(this, activityLogId, environment_id, config) }
-            );
-
-            return this.handleResponse(response, config, activityLogId, environment_id, url);
-        } catch (e: unknown) {
-            return this.handleErrorResponse(e as AxiosError, url, config, activityLogId, environment_id);
-        }
-    }
-
-    /**
-     * Post
-     * @param {Response} res Express response object
-     * @param {NextFuncion} next callback function to pass control to the next middleware function in the pipeline.
-     * @param {string} url
-     * @param {ApplicationConstructedProxyConfiguration} config
-     */
-    private async post(
-        url: string,
+    private async request(
         config: ApplicationConstructedProxyConfiguration,
         activityLogId: number,
         environment_id: number,
         options: AxiosRequestConfig
-    ) {
+    ): Promise<RouteResponse & Activities> {
+        const activityLogs: ActivityLogMessage[] = [];
         try {
-            const headers = this.constructHeaders(config);
             const response: AxiosResponse = await backOff(
                 () => {
-                    return axios.post(url, config.data ?? {}, { ...options, headers });
+                    return axios.request(options);
                 },
-                { numOfAttempts: Number(config.retries), retry: this.retry.bind(this, activityLogId, environment_id, config) }
+                { numOfAttempts: Number(config.retries), retry: this.retry.bind(this, activityLogId, environment_id, config, activityLogs) }
```

Comment:
I am not a huge fan of passing the activityLogs list and mutating it into `retry` but since we cannot get values back from it I don't see a better way. Let me know if you have any suggestion

Reply:
that's the design pattern requirement, I'm good with that